SDL/SDLJoystick.cpp

```diff
@@ -11,167 +11,209 @@ extern "C" {
 	}
 }
 
-SDLJoystick::SDLJoystick(bool init_SDL ): thread(NULL), running(true) {
+SDLJoystick::SDLJoystick(bool init_SDL ): thread(NULL), running(true), player1Controller(NULL), player1JoyInstanceID(-1) {
 	if (init_SDL)
 	{
-		SDL_setenv("SDL_JOYSTICK_ALLOW_BACKGROUND_EVENTS", "1", 0);
-		SDL_Init(SDL_INIT_JOYSTICK | SDL_INIT_VIDEO);
+		SDL_SetHint(SDL_HINT_JOYSTICK_ALLOW_BACKGROUND_EVENTS, "1");
+		SDL_Init(SDL_INIT_JOYSTICK | SDL_INIT_VIDEO | SDL_INIT_GAMECONTROLLER);
 	}
+	
+	if (SDL_GameControllerAddMappingsFromFile("assets/gamecontrollerdb.txt") == -1)
+	{
+			printf("Failed to load control pad mappings, please place gamecontrollerdb.txt in your assets directory\n");
+	}
+	else {
+		printf("Loaded game controller mappings for SDL2\n");
+		setUpPlayer1();
+	}
+}
 
+void SDLJoystick::setUpPlayer1()
+{
 	int numjoys = SDL_NumJoysticks();
-	SDL_JoystickEventState(SDL_ENABLE);
+	int joyIndex = -1;
 	for (int i = 0; i < numjoys; i++) {
-		joys.push_back(SDL_JoystickOpen(i));
-//		printf("Initialized joystick %d: %s",i,SDL_JoystickNameForIndex(i));
-		if(strstr(SDL_JoystickNameForIndex(i),"PLAYSTATION(R)3 Controller"))
-			g_Config.bPS3Controller = true;
+		if (joyIndex == -1 && SDL_IsGameController(i))
+		{
+			joyIndex = i;
+			break;
+		}
+	}
+	if (joyIndex != -1)
+	{
+		printf("Player 1 will be using joystick %d, %s\n", joyIndex, SDL_JoystickNameForIndex(joyIndex));
```

Comment:
There's a gamepad mapping under the Settings which allows you to map controls from specific pads (that's why the deviceId) to certain game input.

For example, you could say that "A" on controller 1 is the PSP's "X", and "A" on controller 2 is the PSP's "O".  Then you could play some weird game with two hands or a friend or whatever weird use case you have.

More practically, you can have multiple controllers simultaneously connected, and choose in the settings which ones map to PSP keys.  People use this for multiplayer netplay AFAIK.

I also think it would be strange if my phone told me it was displaying a webpage on "Display 1".  It only has one display.  That's what I'm saying here: it's confusing for the printf (which many Linux users might see, especially when troubleshooting gamepad issues) to go out of its way to say "Player 1".  Player 1 as opposed to... Player Zed?  Player 9?  There's only one player.

-[Unknown]